File: java_selenium/ThreadQaTests/src/main/java/Pages/WebTablesPage.java
package Pages;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.Select;
import org.openqa.selenium.support.ui.WebDriverWait;
import utils.PropertyReader;

import java.time.Duration;

public class WebTablesPage extends BasePage {

    @FindBy(id = "addNewRecordButton")
    private WebElement addNewRecordButton;

    @FindBy(id = "firstName")
    private WebElement firstName;

    @FindBy(id = "lastName")
    private WebElement lastName;

    @FindBy(id = "userEmail")
    private WebElement userEmail;

    @FindBy(id = "age")
    private WebElement age;

    @FindBy(id = "salary")
    private WebElement salary;

    @FindBy(id = "department")
    private WebElement department;

    @FindBy(id = "submit")
    private WebElement submit;


    public WebTablesPage(WebDriver driver) {
        super(driver);
    }

    public void goToWebTablesPage() {
        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(15));
        driver.get(PropertyReader.getProperty("base_url"));
        wait.until(ExpectedConditions.presenceOfElementLocated(By.xpath("//h5[text()='Elements']")));
        driver.findElement(By.xpath("//h5[text()='Elements']")).click();
        driver.findElement(By.xpath("//span[text()='Web Tables']")).click();
        Select select = new Select(driver.findElement(By.xpath("//select[@aria-label='rows per page']")));
        select.selectByValue("100");
    }

    public void addHumanData(String firstName, String lastName, String userEmail, String age, String salary, String department) {
        addNewRecordButton.click();
        this.firstName.sendKeys(firstName);
        this.lastName.sendKeys(lastName);
        this.userEmail.sendKeys(userEmail);
        this.age.sendKeys(age);
        this.salary.sendKeys(salary);
        this.department.sendKeys(department);
        submit.click();
    }

    public boolean confirmRecord(String firstName, String lastName, String userEmail, String age, String salary, String department) {
        return driver.findElement(By.xpath("//div[text()='" + firstName + "']")) != null &&
                driver.findElement(By.xpath("//div[text()='" + lastName + "']")) != null &&
                driver.findElement(By.xpath("//div[text()='" + userEmail + "']")) != null &&
                driver.findElement(By.xpath("//div[text()='" + age + "']")) != null &&
                driver.findElement(By.xpath("//div[text()='" + salary + "']")) != null &&
                driver.findElement(By.xpath("//div[text()='" + department + "']")) != null;
    }

}

File: python_selenium/ThreadQaTests/page/webtables_page.py
from selenium.common import NoSuchElementException
from selenium.webdriver.common.by import By

from page.base_page import BasePage


class WebTablesPage(BasePage):

    def __init__(self, driver):
        super().__init__(driver)

    def go_to_web_tables_page(self):
        self.driver.get("http://85.192.34.140:8081/")
        self._find_element(By.XPATH, "//h5[text()='Elements']").click()
        self._find_element(By.XPATH, "//span[text()='Web Tables']").click()

    def fill_fields(self, first_name, last_name, email, age, salary, department):
        self._find_element(By.ID, "addNewRecordButton").click()
        self._find_element(By.ID, "firstName").send_keys(first_name)
        self._find_element(By.ID, "lastName").send_keys(last_name)
        self._find_element(By.ID, "userEmail").send_keys(email)
        self._find_element(By.ID, "age").send_keys(age)
        self._find_element(By.ID, "salary").send_keys(salary)
        self._find_element(By.ID, "department").send_keys(department)
        self._find_element(By.ID, "submit").click()

    def confirm_record(self, first_name, last_name, email, age, salary, department):
        try:
            self._find_element(By.XPATH, "//div[text()='" + first_name + "']")
            self._find_element(By.XPATH, "//div[text()='" + last_name + "']")
            self._find_element(By.XPATH, "//div[text()='" + email + "']")
            self._find_element(By.XPATH, "//div[text()='" + age + "']")
            self._find_element(By.XPATH, "//div[text()='" + salary + "']")
            self._find_element(By.XPATH, "//div[text()='" + department + "']")
            return True
        except NoSuchElementException:
            return False
